cpp/benchmarks/common/generate_input.cu

```diff
@@ -250,17 +311,53 @@ auto create_run_length_dist(cudf::size_type avg_run_len)
   return std::gamma_distribution<float>{alpha, avg_run_len / alpha};
 }
 
-// identity mapping, except for bools
-template <typename T, typename Enable = void>
-struct stored_as {
-  using type = T;
-};
-
-// Use `int8_t` for bools because that's how they're stored in columns
-template <typename T>
-struct stored_as<T, std::enable_if_t<std::is_same_v<T, bool>>> {
-  using type = int8_t;
-};
+/**
+ * @brief Generate indices within range [0 , cardinality) repeating with average run length
+ * `avg_run_len`
+ *
+ * @param avg_run_len  Average run length of the generated indices
+ * @param cardinality  Range of the generated indices
+ * @param num_rows     Number of indices to generate
+ * @param engine       Random engine
+ * @return Generated indices of type `cudf::size_type`
+ */
+rmm::device_uvector<cudf::size_type> sample_indices_with_run_length(cudf::size_type avg_run_len,
+                                                                    cudf::size_type cardinality,
+                                                                    cudf::size_type num_rows,
+                                                                    thrust::minstd_rand& engine)
+{
+  auto sample_dist = random_value_fn<cudf::size_type>{
+    distribution_params<cudf::size_type>{distribution_id::UNIFORM, 0, cardinality - 1}};
+  if (avg_run_len > 1) {
+    auto avglen_dist =
+      random_value_fn<int>{distribution_params<int>{distribution_id::UNIFORM, 1, 2 * avg_run_len}};
+    auto approx_run_len = num_rows / avg_run_len + 1;
+    auto run_lens       = avglen_dist(engine, approx_run_len);
+    thrust::inclusive_scan(
+      thrust::device, run_lens.begin(), run_lens.end(), run_lens.begin(), std::plus<int>{});
+    auto samples_indices = sample_dist(engine, approx_run_len + 1);
+    // This is gather.
+    auto avg_repeated_sample_indices = thrust::make_transform_iterator(
```

Comment:
this seems like a wrong name for the iterator.

Reply:
updated.